db/music_pg.sql

```diff
@@ -0,0 +1,373 @@
+CREATE TABLE "submission" (
+  "submission_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "song_id" int NOT NULL,
+  "order" int NOT NULL,
+  "date_submitted" timestamp NOT NULL DEFAULT (now()),
+  "active" bool_int NOT NULL,
+  "is_dupe" bool_int NOT NULL,
+  "title" text NOT NULL,
+  "version" text,
+  "artist" text,
+  "collaborators" text,
+  "composer" text,
+  "link" text NOT NULL,
+  "genre" text,
+  "emoji" text,
+  "comments" text
+);
+
+CREATE TABLE "submission_history" (
+  "submission_history_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "date_update" timestamp NOT NULL,
+  "note_update" text NOT NULL,
+  "submission_id" int NOT NULL,
+  "participant_id" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "song_id" int NOT NULL,
+  "order" int NOT NULL,
+  "date_submitted" timestamp NOT NULL DEFAULT (now()),
+  "active" bool_int NOT NULL,
+  "is_dupe" bool_int NOT NULL,
+  "title" text NOT NULL,
+  "version" text,
+  "artist" text,
+  "collaborators" text,
+  "composer" text,
+  "link" text NOT NULL,
+  "genre" text,
+  "emoji" text,
+  "comments" text
+);
+
+CREATE TABLE "link_relation" (
+  "link_realtion_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "relation_id" int NOT NULL,
+  "link_id" int NOT NULL,
+  "type_link" text NOT NULL
+);
+
+CREATE TABLE "link" (
+  "link_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "url" text NOT NULL,
+  "region" text[]
+);
+
+CREATE TABLE "song" (
+  "song_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "title" text NOT NULL,
+  "title_english" text,
+  "version" text,
+  "artist" text NOT NULL,
+  "collaborators" text,
+  "composer" text,
+  "year_min" int,
+  "year_max" int
+);
+
+CREATE TABLE "song_relation" (
+  "song_relation_id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+  "song_from_id" int NOT NULL,
+  "song_to_id" int NOT NULL,
+  "relation_type" text NOT NULL
+);
+
+CREATE TABLE "song_artist" (
+  "song_artist_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "song_id" int NOT NULL,
+  "artist_id" int NOT NULL,
+  "credit_type" text NOT NULL
+);
+
+CREATE TABLE "artist" (
+  "artist_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text NOT NULL,
+  "year_min" int,
+  "year_max" int,
+  "primary_artist_id" int
+);
+
+CREATE TABLE "artist_genre" (
+  "artist_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "artist_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "participant" (
+  "participant_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "username" text UNIQUE NOT NULL,
+  "nickname" text,
+  "date_joined" timestamp NOT NULL DEFAULT (now()),
+  "time_zone" text,
+  "active" bool_int NOT NULL
+);
+
+CREATE TABLE "participant_nickname" (
+  "participant_nickname_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "nickname" text NOT NULL,
+  "date_start" date NOT NULL,
+  "date_end" date
```

Comment:
I'm not exactly sure how helpful having `date_start` and `date_end` will be here, some people may swap nicknames back and forth a few times, in which case it's hard to keep up with `date_end`. It might be better to just have a `date_start` on there and add a new row every time the nickname doesn't match whatever we think the most recent one is for the given user. This way we'd have multiple records for a single nickname if they swapped into it multiple times, but it would also be obvious which nickname is the most recent in this table just by using `SORT date_start DESC`.

Reply:
It'd be obvious which is current just by finding the one whose date_end is null, though, right? date_end wouldn't have to be when they actually changed, just when they used a new one for something.

Reply:
It just means we'll have to update multiple rows separately when we see a changed nickname instead of just adding a single new one. Having that interdependence between the rows worries me slightly from a data integrity point of view, if our code gets buggy at some point we'll miss something and have two rows with null end date and then we won't know which it is.

Reply:
I'm pretty strongly on the side of having date_end in there. We have a table like this at work, ind_orgs, which relates a user to a company and they did only date_update and it has been a thorn in our sides for years.